src/plot_api/plot_api.js

```diff
@@ -956,6 +956,12 @@ Plotly.addTraces = function addTraces(gd, traces, newIndices) {
     if(!Array.isArray(traces)) {
         traces = [traces];
     }
+
+    // make sure traces do not repeat existing ones
+    traces = traces.map(function(trace) {
+        return Lib.extendFlat({}, trace);
```

Comment:
@dfcreative now that the desired behavior is confirmed. before merging:

TODO

- [ ] rewrite `.map` as for-loop. Good old for-loop are apparently much faster than functional calls at least in Chrome. From our latest estimates in https://github.com/plotly/streambed/pull/8395#discussion_r86844172:

![image](https://cloud.githubusercontent.com/assets/6675409/20275071/0d56f11c-aa65-11e6-8e7d-f96e938da1cf.png)

- [ ] add test case in this [suite](https://github.com/plotly/plotly.js/blob/c597b8cb4514591c50a96535c367ce465037bfae/test/jasmine/tests/plot_api_test.js#L392)